src/core/Block/Block.md

```diff
@@ -1,47 +1,61 @@
-```js
-import { Block } from 'suomifi-ui-components';
+The `<Block>` component is used as a wrapper and spacing component in place of regular HTML elements to enforce correct Suomi.fi styles.
 
-<Block>Oh I'm plain suomifi-div</Block>;
-```
+Examples:
 
-```js
-import { Block } from 'suomifi-ui-components';
-import { createRef } from 'react';
+- [Basic use](/#/Components/Block?id=basic-use)
+- [Spacing](/#/Components/Block?id=spacing)
+- [Semantics](/#/Components/Block?id=semantics)
 
-const exampleRef = createRef();
+<div style="margin-bottom: 40px">
+  [Props & methods](/#/Components/Block?id=props--methods)
+</div>
 
-<Block ref={exampleRef}>Block with ref</Block>;
-```
+### Basic use
 
 ```js
 import { Block } from 'suomifi-ui-components';
 
-<Block padding="xxxl" style={{ border: '1px solid red' }}>
-  Block with xl-padding
-</Block>;
+<Block>I'm a div with Suomi.fi styles</Block>;
 ```
 
+### Spacing
+
+You can create spacing around the `<Block>` component using the margin and padding props. Refer to [Props & methods](/#/Components/Block?id=props--methods) for the full list.
+
 ```js
 import { Block } from 'suomifi-ui-components';
 
-<Block
-  pt="s"
-  pr="l"
-  pb="xxl"
-  pl="xxxl"
-  style={{ border: '1px solid red' }}
->
-  Block with indepedent paddings on each side
-</Block>;
+<>
+  <Block padding="xxxl" style={{ border: '1px solid red' }}>
+    Block with xl-padding
+  </Block>
+  <Block
+    pt="s"
+    pr="l"
+    pb="xxl"
+    pl="xxxl"
+    style={{ border: '1px solid red' }}
+  >
+    Block with indepedent paddings on each side
+  </Block>
+</>;
 ```
 
+### Semantics
+
+The `<Block>` component can be rendered as a variety of different HTML elements. Refer to props & methods for the full list.
+
 ```js
 import { Block } from 'suomifi-ui-components';
 
 <>
-  <Block variant="section">I'm semantically a section</Block>
-  <Block variant="span">
+  <Block variant="section" style={{ border: '1px solid red' }}>
```

Comment:
Should we mention here that `section`'s should have an aria-label (or aria-labelledby) to show up as a landmark. Using sections on their own has very little semantic meaning for most assistive technology. They are equivalent to using the generic ARIA role "region".

Reply:
I would say this page is documentation for the <Block> component's features and not an instruction in terms of HTML element usage. I changed the example to have a `<Block variant="footer">` instead to not have such a bad example of section usage :)